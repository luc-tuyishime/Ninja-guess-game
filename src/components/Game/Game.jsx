import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import '../../assets/css/style.scss';
import './Game.scss';
import Profile from '../Profile/Profile';
import Quiz from './Quiz/Quiz';
import Members from '../Game/Members/Members';

class Game extends Component {
  render() {
    const {
      match: {
        params: { roomName }
      }
    } = this.props;
    return (
      <div className="container">
        <div className="small-screen-4 medium-screen-1 large-screen-1">
          <Profile roomName={roomName} />
        </div>
        {roomName ? (
          <div className="small-screen-4 medium-screen-3 large-screen-3 ">
            <div>
              <Quiz roomName={roomName} />
            </div>
            <div className="row">
              <div className="card radius-2">
                <Members roomName={roomName} />
                <div className="divider" />
              </div>
            </div>
          </div>
        ) : null}
      </div>
    );
  }
}

Game.propTypes = {};

const mapStateToProps = ({ user: { profile } }) => ({
  profile
});
export const mapDispatchToProps = dispatch => {};

export default connect(
  mapStateToProps,
  null
)(Game);
