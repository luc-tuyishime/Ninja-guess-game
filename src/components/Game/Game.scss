@import '../../assets/css/_colors';
@import '../../assets/css/_pixels';

.grabGame {
  margin-top: 25px;
}
.nameAdded {
  border: #a7a7a8 1px solid;
  padding: 20px;
  margin: 20px 0px;
  border-radius: 8px;
}

.submit {
  padding-top: 15px !important;
  padding-bottom: 15px !important;
  margin-left: 15px !important;
  margin-top: 15px;
}

.oneResult {
  padding: 15px 1%;
  margin: 3px 0.5%;
  width: 46%;
  float: left;
}
.oneResult span {
  float: right;
}

.correct {
  background: olivedrab !important;
}
.wrong {
  background: $danger;
}
