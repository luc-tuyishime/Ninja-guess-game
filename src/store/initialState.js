const user = require('../helpers/checkUser')();

module.exports = {
  game: {
    room: {},
    geography: {
      countries: [
        { name: 'Abkhazia', capital: 'Sukhumi' },
        { name: 'Afghanistan', capital: 'Kabul' },
        { name: 'Akrotiri and Dhekelia', capital: 'Episkopi Cantonment' },
        { name: 'Albania', capital: 'Tirana' },
        { name: 'Algeria', capital: 'Algiers' },
        { name: 'American Samoa', capital: 'Pago Pago' },
        { name: 'Andorra', capital: 'Andorra la Vella' },
        { name: 'Angola', capital: 'Luanda' },
        { name: 'Anguilla', capital: 'The Valley' },
        { name: 'Antigua and Barbuda', capital: "St. John's" },
        { name: 'Argentina', capital: 'Buenos Aires' },
        { name: 'Armenia', capital: 'Yerevan' },
        { name: 'Aruba', capital: 'Oranjestad' },
        { name: 'Ascension Island', capital: 'Georgetown' },
        { name: 'Australia', capital: 'Canberra' },
        { name: 'Austria', capital: 'Vienna' },
        { name: 'Azerbaijan', capital: 'Baku' },
        { name: 'Bahamas', capital: 'Nassau' },
        { name: 'Bahrain', capital: 'Manama' },
        { name: 'Bangladesh', capital: 'Dhaka' },
        { name: 'Barbados', capital: 'Bridgetown' },
        { name: 'Belarus', capital: 'Minsk' },
        { name: 'Belgium', capital: 'Brussels' },
        { name: 'Belize', capital: 'Belmopan' },
        { name: 'Benin', capital: 'Porto-Novo' },
        { name: 'Bermuda', capital: 'Hamilton' },
        { name: 'Bhutan', capital: 'Thimphu' },
        { name: 'Bolivia', capital: 'Sucre' },
        { name: 'Bolivia', capital: 'La Paz' },
        { name: 'Bosnia and Herzegovina', capital: 'Sarajevo' },
        { name: 'Botswana', capital: 'Gaborone' },
        { name: 'Brazil', capital: 'Brasília' },
        { name: 'British Virgin Islands', capital: 'Road Town' },
        { name: 'Brunei', capital: 'Bandar Seri Begawan' },
        { name: 'Bulgaria', capital: 'Sofia' },
        { name: 'Burkina Faso', capital: 'Ouagadougou' },
        { name: 'Burundi', capital: 'Bujumbura' },
        { name: 'Cambodia', capital: 'Phnom Penh' },
        { name: 'Cameroon', capital: 'Yaoundé' },
        { name: 'Canada', capital: 'Ottawa' },
        { name: 'Cape Verde', capital: 'Praia' },
        { name: 'Cayman Islands', capital: 'George Town' },
        { name: 'Central African Republic', capital: 'Bangui' },
        { name: 'Chad', capital: "N'Djamena" },
        { name: 'Chile', capital: 'Santiago' },
        { name: 'China', capital: 'Beijing' },
        { name: 'Christmas Island', capital: 'Flying Fish Cove' },
        { name: 'Cocos (Keeling) Islands', capital: 'West Island' },
        { name: 'Colombia', capital: 'Bogotá' },
        { name: 'Comoros', capital: 'Moroni' },
        { name: 'Cook Islands', capital: 'Avarua' },
        { name: 'Costa Rica', capital: 'San José' },
        { name: 'Croatia', capital: 'Zagreb' },
        { name: 'Cuba', capital: 'Havana' },
        { name: 'Curaçao', capital: 'Willemstad' },
        { name: 'Cyprus', capital: 'Nicosia' },
        { name: 'Czech Republic', capital: 'Prague' },
        { name: "Côte d'Ivoire", capital: 'Yamoussoukro' },
        { name: 'Democratic Republic of the Congo', capital: 'Kinshasa' },
        { name: 'Denmark', capital: 'Copenhagen' },
        { name: 'Djibouti', capital: 'Djibouti' },
        { name: 'Dominica', capital: 'Roseau' },
        { name: 'Dominican Republic', capital: 'Santo Domingo' },
        { name: 'East Timor (Timor-Leste)', capital: 'Dili' },
        { name: 'Easter Island', capital: 'Hanga Roa' },
        { name: 'Ecuador', capital: 'Quito' },
        { name: 'Egypt', capital: 'Cairo' },
        { name: 'El Salvador', capital: 'San Salvador' },
        { name: 'Equatorial Guinea', capital: 'Malabo' },
        { name: 'Eritrea', capital: 'Asmara' },
        { name: 'Estonia', capital: 'Tallinn' },
        { name: 'Ethiopia', capital: 'Addis Ababa' },
        { name: 'Falkland Islands', capital: 'Stanley' },
        { name: 'Faroe Islands', capital: 'Tórshavn' },
        { name: 'Federated States of Micronesia', capital: 'Palikir' },
        { name: 'Fiji', capital: 'Suva' },
        { name: 'Finland', capital: 'Helsinki' },
        { name: 'France', capital: 'Paris' },
        { name: 'French Guiana', capital: 'Cayenne' },
        { name: 'French Polynesia', capital: 'Papeete' },
        { name: 'Gabon', capital: 'Libreville' },
        { name: 'Gambia', capital: 'Banjul' },
        { name: 'Georgia', capital: 'Tbilisi' },
        { name: 'Germany', capital: 'Berlin' },
        { name: 'Ghana', capital: 'Accra' },
        { name: 'Gibraltar', capital: 'Gibraltar' },
        { name: 'Greece', capital: 'Athens' },
        { name: 'Greenland', capital: 'Nuuk' },
        { name: 'Grenada', capital: "St. George's" },
        { name: 'Guam', capital: 'Hagåtña' },
        { name: 'Guatemala', capital: 'Guatemala City' },
        { name: 'Guernsey', capital: 'St. Peter Port' },
        { name: 'Guinea', capital: 'Conakry' },
        { name: 'Guinea-Bissau', capital: 'Bissau' },
        { name: 'Guyana', capital: 'Georgetown' },
        { name: 'Haiti', capital: 'Port-au-Prince' },
        { name: 'Honduras', capital: 'Tegucigalpa' },
        { name: 'Hungary', capital: 'Budapest' },
        { name: 'Iceland', capital: 'Reykjavík' },
        { name: 'India', capital: 'New Delhi' },
        { name: 'Indonesia', capital: 'Jakarta' },
        { name: 'Iran', capital: 'Tehran' },
        { name: 'Iraq', capital: 'Baghdad' },
        { name: 'Ireland', capital: 'Dublin' },
        { name: 'Isle of Man', capital: 'Douglas' },
        { name: 'Israel', capital: 'Jerusalem' },
        { name: 'Italy', capital: 'Rome' },
        { name: 'Jamaica', capital: 'Kingston' },
        { name: 'Japan', capital: 'Tokyo' },
        { name: 'Jersey', capital: 'St. Helier' },
        { name: 'Jordan', capital: 'Amman' },
        { name: 'Kazakhstan', capital: 'Astana' },
        { name: 'Kenya', capital: 'Nairobi' },
        { name: 'Kiribati', capital: 'Tarawa' },
        { name: 'Kosovo', capital: 'Pristina' },
        { name: 'Kuwait', capital: 'Kuwait City' },
        { name: 'Kyrgyzstan', capital: 'Bishkek' },
        { name: 'Laos', capital: 'Vientiane' },
        { name: 'Latvia', capital: 'Riga' },
        { name: 'Lebanon', capital: 'Beirut' },
        { name: 'Lesotho', capital: 'Maseru' },
        { name: 'Liberia', capital: 'Monrovia' },
        { name: 'Libya', capital: 'Tripoli' },
        { name: 'Liechtenstein', capital: 'Vaduz' },
        { name: 'Lithuania', capital: 'Vilnius' },
        { name: 'Luxembourg', capital: 'Luxembourg' },
        { name: 'Macedonia', capital: 'Skopje' },
        { name: 'Madagascar', capital: 'Antananarivo' },
        { name: 'Malawi', capital: 'Lilongwe' },
        { name: 'Malaysia', capital: 'Kuala Lumpur' },
        { name: 'Maldives', capital: 'Malé' },
        { name: 'Mali', capital: 'Bamako' },
        { name: 'Malta', capital: 'Valletta' },
        { name: 'Marshall Islands', capital: 'Majuro' },
        { name: 'Mauritania', capital: 'Nouakchott' },
        { name: 'Mauritius', capital: 'Port Louis' },
        { name: 'Mexico', capital: 'Mexico City' },
        { name: 'Moldova', capital: 'Chisinau' },
        { name: 'Monaco', capital: 'Monaco' },
        { name: 'Mongolia', capital: 'Ulaanbaatar' },
        { name: 'Montenegro', capital: 'Podgorica' },
        { name: 'Montserrat', capital: 'Plymouth' },
        { name: 'Morocco', capital: 'Rabat' },
        { name: 'Mozambique', capital: 'Maputo' },
        { name: 'Myanmar', capital: 'Naypyidaw' },
        { name: 'Nagorno-Karabakh Republic', capital: 'Stepanakert' },
        { name: 'Namibia', capital: 'Windhoek' },
        { name: 'Nauru', capital: 'Yaren' },
        { name: 'Nepal', capital: 'Kathmandu' },
        { name: 'Netherlands', capital: 'Amsterdam' },
        { name: 'New Caledonia', capital: 'Nouméa' },
        { name: 'New Zealand', capital: 'Wellington' },
        { name: 'Nicaragua', capital: 'Managua' },
        { name: 'Niger', capital: 'Niamey' },
        { name: 'Nigeria', capital: 'Abuja' },
        { name: 'Niue', capital: 'Alofi' },
        { name: 'Norfolk Island', capital: 'Kingston' },
        { name: 'North Korea', capital: 'Pyongyang' },
        { name: 'Northern Cyprus', capital: 'Nicosia' },
        { name: 'United Kingdom Northern Ireland', capital: 'Belfast' },
        { name: 'Northern Mariana Islands', capital: 'Saipan' },
        { name: 'Norway', capital: 'Oslo' },
        { name: 'Oman', capital: 'Muscat' },
        { name: 'Pakistan', capital: 'Islamabad' },
        { name: 'Palau', capital: 'Ngerulmud' },
        { name: 'Palestine', capital: 'Jerusalem' },
        { name: 'Panama', capital: 'Panama City' },
        { name: 'Papua New Guinea', capital: 'Port Moresby' },
        { name: 'Paraguay', capital: 'Asunción' },
        { name: 'Peru', capital: 'Lima' },
        { name: 'Philippines', capital: 'Manila' },
        { name: 'Pitcairn Islands', capital: 'Adamstown' },
        { name: 'Poland', capital: 'Warsaw' },
        { name: 'Portugal', capital: 'Lisbon' },
        { name: 'Puerto Rico', capital: 'San Juan' },
        { name: 'Qatar', capital: 'Doha' },
        { name: 'Republic of China (Taiwan)', capital: 'Taipei' },
        { name: 'Republic of the Congo', capital: 'Brazzaville' },
        { name: 'Romania', capital: 'Bucharest' },
        { name: 'Russia', capital: 'Moscow' },
        { name: 'Rwanda', capital: 'Kigali' },
        { name: 'Saint Barthélemy', capital: 'Gustavia' },
        { name: 'Saint Helena', capital: 'Jamestown' },
        { name: 'Saint Kitts and Nevis', capital: 'Basseterre' },
        { name: 'Saint Lucia', capital: 'Castries' },
        { name: 'Saint Martin', capital: 'Marigot' },
        { name: 'Saint Pierre and Miquelon', capital: 'St. Pierre' },
        { name: 'Saint Vincent and the Grenadines', capital: 'Kingstown' },
        { name: 'Samoa', capital: 'Apia' },
        { name: 'San Marino', capital: 'San Marino' },
        { name: 'Saudi Arabia', capital: 'Riyadh' },
        { name: 'Scotland', capital: 'Edinburgh' },
        { name: 'Senegal', capital: 'Dakar' },
        { name: 'Serbia', capital: 'Belgrade' },
        { name: 'Seychelles', capital: 'Victoria' },
        { name: 'Sierra Leone', capital: 'Freetown' },
        { name: 'Singapore', capital: 'Singapore' },
        { name: 'Sint Maarten', capital: 'Philipsburg' },
        { name: 'Slovakia', capital: 'Bratislava' },
        { name: 'Slovenia', capital: 'Ljubljana' },
        { name: 'Solomon Islands', capital: 'Honiara' },
        { name: 'Somalia', capital: 'Mogadishu' },
        { name: 'Somaliland', capital: 'Hargeisa' },
        { name: 'South Africa', capital: 'Pretoria' },
        {
          name: 'South Georgia and the South Sandwich Islands',
          capital: 'Grytviken'
        },
        { name: 'South Korea', capital: 'Seoul' },
        { name: 'South Ossetia', capital: 'Tskhinvali' },
        { name: 'South Sudan South Sudan', capital: 'Juba' },
        { name: 'Spain', capital: 'Madrid' },
        { name: 'Sri Lanka', capital: 'Sri Jayawardenapura Kotte' },
        { name: 'Sudan', capital: 'Khartoum' },
        { name: 'Suriname', capital: 'Paramaribo' },
        { name: 'Swaziland', capital: 'Mbabane' },
        { name: 'Sweden', capital: 'Stockholm' },
        { name: 'Switzerland', capital: 'Bern' },
        { name: 'Syria', capital: 'Damascus' },
        { name: 'São Tomé and Príncipe', capital: 'São Tomé' },
        { name: 'Tajikistan', capital: 'Dushanbe' },
        { name: 'Tanzania', capital: 'Dodoma' },
        { name: 'Thailand', capital: 'Bangkok' },
        { name: 'Togo', capital: 'Lomé' },
        { name: 'Tonga', capital: 'Nukuʻalofa' },
        { name: 'Transnistria', capital: 'Tiraspol' },
        { name: 'Trinidad and Tobago', capital: 'Port of Spain' },
        { name: 'Tristan da Cunha', capital: 'Edinburgh of the Seven Seas' },
        { name: 'Tunisia', capital: 'Tunis' },
        { name: 'Turkey', capital: 'Ankara' },
        { name: 'Turkmenistan', capital: 'Ashgabat' },
        { name: 'Turks and Caicos Islands', capital: 'Cockburn Town' },
        { name: 'Tuvalu', capital: 'Funafuti' },
        { name: 'Uganda', capital: 'Kampala' },
        { name: 'Ukraine', capital: 'Kiev' },
        { name: 'United Arab Emirates', capital: 'Abu Dhabi' },
        { name: 'United Kingdom; England', capital: 'London' },
        { name: 'United States', capital: 'Washington, D.C.' },
        { name: 'United States Virgin Islands', capital: 'Charlotte Amalie' },
        { name: 'Uruguay', capital: 'Montevideo' },
        { name: 'Uzbekistan', capital: 'Tashkent' },
        { name: 'Vanuatu', capital: 'Port Vila' },
        { name: 'Vatican City', capital: 'Vatican City' },
        { name: 'Venezuela', capital: 'Caracas' },
        { name: 'Vietnam', capital: 'Hanoi' },
        { name: 'Wales', capital: 'Cardiff' },
        { name: 'Wallis and Futuna', capital: 'Mata-Utu' },
        { name: 'Western Sahara', capital: 'El Aaiún' },
        { name: 'Yemen', capital: 'Sanaá' },
        { name: 'Zambia', capital: 'Lusaka' },
        { name: 'Zimbabwe', capital: 'Harame' }
      ]
    },
    computing: {
      programming: [
        { term: 'OOP', definition: 'object oriented programming' },
        {
          term: 'polymorphism',
          definition: 'the ability of an object to take on many forms'
        },
        {
          term: 'inheritance',
          definition:
            'a mechanism in which one class acquires the property of another class'
        },
        {
          term: 'state',
          definition:
            'an instance of React Component Class can be defined as an object of a set of observable properties that control the behaviour of the component'
        },
        {
          term: 'component',
          definition: 'the building blocks of any React app'
        }
      ]
    },
    answer: {},
    attempts: {
      geography: [],
      computing: []
    },
    results: [],
    members: []
  },
  user: {
    profile: user.profile,
    isAuth: user.isAuth,
    errors: null
  }
};
