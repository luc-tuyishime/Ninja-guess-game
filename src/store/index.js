import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import initialState from './initialState';
import reducers from '../reducers';

const hostname = window.location.hostname;

export default createStore(
  reducers,
  initialState,
  hostname === 'localhost'
    ? composeWithDevTools(applyMiddleware(thunk))
    : applyMiddleware(thunk)
);
