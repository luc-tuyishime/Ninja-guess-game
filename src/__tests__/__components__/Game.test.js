import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import Game from '../../components/Game/Game';
import { Provider } from 'react-redux';
import store from '../../store';

describe('Game', () => {
  test('game', () => {
    const component = create(
      <Provider store={store}>
        <Game />
      </Provider>
    );
    const rootInstance = component.root;
    const form = rootInstance.findByType('form');
    form.props.onSubmit({
      preventDefault: () => console.log('preventDefault'),
      target: {
        answer: {
          value: 'Kigali'
        }
      }
    });
    // expect(component.toJSON()).toMatchSnapshot();
  });
  // test('home page (/home)', () => {
  //   const component = create(
  //     <Provider store={store}>
  //       <MemoryRouter initialEntries={['/home']}>
  //         <Game />
  //       </MemoryRouter>
  //     </Provider>
  //   );
  //   expect(component.toJSON()).toMatchSnapshot();
  // });

  // test('game page (/game)', () => {
  //   const component = create(
  //     <Provider store={store}>
  //       <MemoryRouter initialEntries={['/game']}>
  //         <Game />
  //       </MemoryRouter>
  //     </Provider>
  //   );
  //   expect(component.toJSON()).toMatchSnapshot();
  // });
});
