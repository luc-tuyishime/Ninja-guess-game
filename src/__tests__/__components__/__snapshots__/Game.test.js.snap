// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game game 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <div
      className="row shadow-3 radius-4 "
    >
      <div
        className="grabProfileImage center"
      >
        <img
          alt="Profile"
          className="center shadow-4"
          src="profile_plaholder.png"
        />
      </div>
      <div
        className="card no-margin center-align"
      >
        <div
          className="text-info large-text"
        >
          Hello,
          <br />
           
        </div>
         
        <h1
          className="no-margin"
        >
           
        </h1>
      </div>
      <div
        className="clear"
      />
      <div
        className="container"
      >
        <div
          className="card center-align"
        >
          <button
            className="radius-4 button danger radius-5 text-white large-h-padding shadow-3 large-text"
            onClick={[Function]}
            type="button"
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container shadow-2 radius-3"
  >
    <div
      className="center"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="primary card radius-3 bold large-text text-white center-align"
          id="question"
        >
          What is the capital of Sierra Leone ?
        </div>
        <div
          className="hide"
          id="country"
        >
          Sierra Leone
        </div>
        <div
          className="radius-2 input-field"
        >
          <div
            className="card"
          >
            <input
              autoComplete="off"
              className="shadow-4 radius-2 medium-padding card medium-text large-screen-3"
              id="name"
              name="answer"
              placeholder="Enter your answer"
              type="text"
            />
          </div>
          <br />
          <div
            className="card"
          >
            <button
              className="submit radius-4 button radius-5 primary text-white center-align large-h-padding large-text"
              type="submit"
            >
              Answer
            </button>
            <button
              className="submit radius-4 button radius-5 primary text-white center-align large-h-padding large-text"
              onClick={[Function]}
            >
              Skip
            </button>
          </div>
        </div>
      </form>
      <div
        className="grey container shadow-2 radius-3"
      >
        
      </div>
    </div>
  </div>
</div>
`;
